### PR TITLE
fix(canendar): subtitle supports reactive

### LINKED ISSUE #12423: [Bug Report] Locale.use 无法动态修改 Calendar 组件的 subtitle
### 重现链接

https://codesandbox.io/s/vant-4-issue-template-forked-wvj6kx?file=/src/App.vue

### Vant 版本

4.7.3

### 描述一下你遇到的问题。

Locale.use 无法动态修改 Calendar 组件的 subtitle

### 重现步骤

如果组件初始化之前就设置为英语则会显示成 2023/11

组件初始化之后使用 Locale.use('en-US’, enUS) 则不能动态切换

### 设备/浏览器

_No response_

### PR DESCRIPTION
**Pull request by cc-hearts**
fix: subTitle supports responsive switching when international language changes

close #12423

### CODE REVIEW

File: packages/vant/src/calendar/CalendarMonth.tsx [hunk 1]
```diff
@@ -87,7 +88,7 @@ export default defineComponent({
 
     const shouldRender = computed(() => visible.value || !props.lazyRender);
 
-    const getTitle = () => title.value;
+    const getTitle = () => toRef(title, 'value');
```
Comment (chenjiahan) [inline packages/vant/src/calendar/CalendarMonth.tsx:91]: Is this a breaking change? The `title` param of `month-show` event is changed.

The previous type was string, but now the type is Ref<string>

Comment (cc-hearts) [reply packages/vant/src/calendar/CalendarMonth.tsx:91]: this is a  breaking change. the `title` should change responsively with `locale` changes.

The `title` type didn't change, that was my mistake.

**Comment by cc-hearts** [comment]
🤯 I did a lot of breaking change.

For `subTitle` assignments, use `shallowReactive` instead of `ref`. This way `subTitle.text` can be declared using `ref`.

**Comment by chenjiahan** [comment]
We can not introduce any breaking changes in a minor or patch release, so we can only make forward-compatible changes.

**Comment by cc-hearts** [comment]
Thanks to your suggestion, I removed breaking change.

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
Thanks!

**Pull request merged**